include/zephyr/net/coap.h

```diff
@@ -1164,6 +1164,36 @@ int coap_resource_notify(struct coap_resource *resource);
  */
 bool coap_request_is_observe(const struct coap_packet *request);
 
+/**
+ * @brief Get number of retransmissions for CoAP packet.
+ *
+ * @return Maximum number of retransmissions.
+ */
+uint8_t coap_get_max_retransmission(void);
+
+/**
+ * @brief Set maximum number of retransmissions.
+ *
+ * @param retries Number of retransmissions.
+ */
+void coap_set_max_retransmission(uint8_t retries);
+
+/**
+ * @brief Set initial AKC timeout. Value is used as a base value
+ * to retry pending CoAP packets.
+ *
+ * @param timeout Timeout in milliseconds.
+ */
+void coap_set_ack_timeout(uint32_t timeout);
+
+/**
+ * @brief Set CoAP retry backoff factor. A value of 200 means
+ * a factor of 2.0.
+ *
+ * @param percent Backoff factor.
+ */
+void coap_set_backoff_percent(uint16_t percent);
+
```

Comment:
I would prefer that there is just one configuration struct, and one set of get/set functions

```suggestion
/**
 * @brief Coap transmission configuration
 */
struct coap_config_t {
    uint8_t max_retransmission; /**< Maximum number of retransmissions. */
    uint32_t ack_timeout;   /**<  initial AKC timeout. Value is used as a base value to retry pending CoAP packets. */
    uint16_t coap_backoff_percent; /**< Set CoAP retry backoff factor. A value of 200 means a factor of 2.0. */
 }

/**
 * @brief Get currently active CoAP configuration.
 *
 * @return configuration.
 */
struct coap_config_t coap_get_config(void);

/**
 * @brief Set CoAP configuration.
 *
 * @param conf Configuration structure
 */
void coap_set_config(struct coap_config_t conf);


```

Reply:
Perhaps we could even add a pointer to such a struct from the `coap_packet` struct to allow the caller control over the behavior for individual requests.
If this pointer is `NULL`, fallback to the defaults.

Just thinking out loud here.